.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  ci:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: [14]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Setup node env
+        uses: actions/setup-node@v2.1.2
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Cache node_modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```

Comment:
why do you need this here?

Reply:
that is for caching node_modules. Is that not something we want?

Reply:
well `key` is already used for cacheing, I'm just wondering why you want `restore-keys`

Reply:
It's so that it'll match with a generic key without the hash in case there was a cache miss using the key

Reply:
when will the cache be populated with a generic key?